### PR TITLE
Use cw-storage-plus

### PR DESCRIPTION
**Pull request by orkunkl**
Better to make plus storage visible to newbies.

### CODE REVIEW

**Review by webmaster128 — COMMENTED** [review]
This brings us into dependency hell. For every new version of CosmWasm, we need to upgrade the whole cosmwasm-plus repo before being able to update the template.

**Comment by orkunkl** [comment]
If plus-storage matured enough, maybe it is time to put it under separate repo?

**Comment by ethanfrey** [comment]
> This brings us into dependency hell. For every new version of CosmWasm, we need to upgrade the whole cosmwasm-plus repo before being able to update the template.

Well, we are usually doing `-alpha`or `-beta` releases, so we can easily make a PR that is working with eg. cosmwasm-plus `v0.7.0-beta1` and test that. And as soon as we tag cosmwasm-plus `v0.7.0` we can update and merge cosmwasm-template.

cosmwasm-plus is the most maintained repo outside of cosmwasm (cosmwasm-template is usually an afterthought and cosmwasm-examples more or less forgotten), so I don't see an issue with importing it.

File: src/state.rs [hunk 1]
```diff
@@ -4,18 +4,10 @@ use serde::{Deserialize, Serialize};
...
 }
 
-pub fn config(storage: &mut dyn Storage) -> Singleton<State> {
-    singleton(storage, CONFIG_KEY)
-}
-
-pub fn config_read(storage: &dyn Storage) -> ReadonlySingleton<State> {
-    singleton_read(storage, CONFIG_KEY)
-}
+const STATE: Item<State> = Item::new("state");
```
Comment (ethanfrey) [inline src/state.rs:13]: It is nice to make this idiom more visibile. We should clarify the dependency issue first however

**Comment by webmaster128** [comment]
Alright, fine with me. I just want to raise awareness of this problem, which hit us in CosmJS already where the generic CosmWasm client suddenly depended on cosmwasm-plus. If the benefics are greater than the pain, go for it.

**Comment by ethanfrey** [comment]
I understand the pain. And cosmwasm-plus was lagging on a few upgrades. But we are now keeping it quite up to date and I doubt there will be multi-day PRs needed to update the dependencies any more.

**Review by ethanfrey — COMMENTED** [review]
Code looks good to me, but imports are broken: https://ci.appveyor.com/project/confio/cosmwasm-template/builds/39191175

The way I deal with this is:
* cargo generate a project from a local branch
* cd to the new project
* git commit that inital state
* try to `cargo build` and `cargo test` until it passes
* use `git diff` to see the changes and "backport" them to this repo.

At that point the CI should pass.

File: Developing.md [hunk 2]
```diff
@@ -3,12 +3,12 @@
 If you have recently created a contract with this template, you probably could use some
 help on how to build and test the contract, as well as prepare it for production. This
 file attempts to provide a brief overview, assuming you have installed a recent
-version of Rust already (eg. 1.47.0+).
+version of Rust already (eg. 1.51.0+).
```
Comment (orkunkl) [inline Developing.md:6]: Just noticed these and fixed it on the way.

**Pull request merged**